octodns_azure/__init__.py

```diff
@@ -594,8 +603,14 @@ def _check_zone(self, name, create=False):
         if create:
             self.log.debug('_check_zone:no matching zone; creating %s', name)
             create_zone = self._dns_client.zones.create_or_update
-            create_zone(self._resource_group, name, Zone(location='global'))
+            zone = create_zone(
+                self._resource_group, name, Zone(location='global'))
             self._azure_zones.add(name)
+
+            # we create the zone so we should now be able to get its root ns
+            # records
+            self._required_root_ns_values[name] = set(zone.name_servers)
```

Comment:
Found an even cleaner way. The nameserver list is embedded within the Zone object itself 🤷

Reply:
Oh. Nice. That definitely works.